### ISSUE TITLE
Outputfilter does not return pure HTML anymore when using nocache tags in included templates

### ISSUE THREAD

**Issue by kkFelix** [smarty-php/smarty#160]
Minimal example:
```
function opf($tpl_output, Smarty_Internal_Template $template) {
    echo '<pre>';
    var_dump(htmlentities($tpl_output));
    echo '</pre>';

    return $tpl_output . '<br>Text from filter';
}

$smarty = new Smarty();
$smarty->setCompileDir('/tmp/smarty/compile');
$smarty->setCacheDir('/tmp/smarty/cache');
$smarty->setCaching(true);
$smarty->setTemplateDir('/tmp/smarty/tpl');
$smarty->registerFilter('output', 'opf');
$smarty->display('test.tpl');
```
test.tpl:
```
{include file=test2.tpl}
```
test2.tpl:
```
{nocache}
<p>Timestamp: {$smarty.now}</p>
{/nocache}
```

In Smarty 3.1.27 (or without caching) the $tpl_output variable in the output filter function would be something like
```
<p>Timestamp: 1452604029</p>
```
In 3.1.29 (and possibly .28) with enabled cache it is instead 
```
<p>Timestamp: <?php echo time();?></p>
```

I think it's because of Smarty_Internal_Runtime_UpdateCache::removeNoCacheHash() but did not investigate further.

**Comment by uwetews** [comment]
I can confirm that this bug does exists in 3.1.29. 
However some of the rendering code has been rewritten for the upcoming version 3.1.30.
This does also solve your problem.

3.1.30 will be released after I have solved a couple of other open issues.
You may try the master branch version to get the latest updates.

**Comment by uwetews** [comment]
The bugfix is now in the master branch and will later be included in 3.1.30
